dsgrid/utils/spark.py

```diff
@@ -407,3 +408,102 @@ def _get_spark_session():
         spark = SparkSession.builder.getOrCreate()
         log_spark_conf(spark)
     return spark
+
+
+def try_load_stored_table(table_name, database="default"):
+    """Return a table stored in the Spark warehouse.
+
+    Parameters
+    ----------
+    table_name : str
+    database : str, optional
+        database, by default "default"
+
+    Returns
+    -------
+    pyspark.sql.DataFrame
+
+    """
+    spark = _get_spark_session()
+    if spark.catalog.tableExists(table_name, dbName=database):
+        return spark.table(table_name)
+    table = try_load_stored_table_from_warehouse(table_name)
+    if table is not None:
+        logger.warning("Table %s was in warehouse but not spark catalog", table_name)
+    return table
+
+
+def is_table_stored(table_name, database="default"):
+    return table_name in list_tables(database=database)
```

Comment:
I assume spark_db and spark warehouse are two different things and this check whether it's stored in either (as opposed to `_is_table_stored_in_spark_db`) ?

Reply:
Based on the documentation you should be able to reuse tables across sessions, like this
```
$ pyspark
In [1]: spark.createDataFrame([{"a": 1}]).write.saveAsTable("my_table")
In [2]: exit()
$ pyspark
In [1]: spark.sql("select * from my_table")
```
It does work for me generally, but not always within dsgrid and I don't know why. I gave up debugging and just started checking in the warehouse directory manually. Refer to line 471